libdispatch/CMakeLists.txt

```diff
@@ -4,57 +4,62 @@
 # University Corporation for Atmospheric Research/Unidata.
 
 # See netcdf-c/COPYRIGHT file for more info.
-SET(libdispatch_SOURCES dcopy.c dfile.c ddim.c datt.c dattinq.c dattput.c dattget.c derror.c dvar.c dvarget.c dvarput.c dvarinq.c ddispatch.c nclog.c dstring.c dutf8.c dinternal.c doffsets.c ncuri.c nclist.c ncbytes.c nchashmap.c nctime.c nc.c nclistmgr.c utf8proc.h utf8proc.c dpathmgr.c dutil.c drc.c dauth.c dreadonly.c dnotnc4.c dnotnc3.c dinfermodel.c
+set(libdispatch_SOURCES dcopy.c dfile.c ddim.c datt.c dattinq.c dattput.c dattget.c derror.c dvar.c dvarget.c dvarput.c dvarinq.c ddispatch.c nclog.c dstring.c dutf8.c dinternal.c doffsets.c ncuri.c nclist.c ncbytes.c nchashmap.c nctime.c nc.c nclistmgr.c utf8proc.h utf8proc.c dpathmgr.c dutil.c drc.c dauth.c dreadonly.c dnotnc4.c dnotnc3.c dinfermodel.c
 daux.c dinstance.c dinstance_intern.c
 dcrc32.c dcrc32.h dcrc64.c ncexhash.c ncxcache.c ncjson.c ds3util.c dparallel.c dmissing.c)
 
 # Netcdf-4 only functions. Must be defined even if not used
-SET(libdispatch_SOURCES ${libdispatch_SOURCES} dgroup.c dvlen.c dcompound.c dtype.c denum.c dopaque.c dfilter.c)
+set(libdispatch_SOURCES ${libdispatch_SOURCES} dgroup.c dvlen.c dcompound.c dtype.c denum.c dopaque.c dfilter.c)
 
-IF(BUILD_V2)
-  SET(libdispatch_SOURCES ${libdispatch_SOURCES} dv2i.c)
-ENDIF(BUILD_V2)
+if(BUILD_V2)
+  set(libdispatch_SOURCES ${libdispatch_SOURCES} dv2i.c)
+endif(BUILD_V2)
 
-IF(ENABLE_BYTERANGE)
-  SET(libdispatch_SOURCES ${libdispatch_SOURCES} dhttp.c)
+if(ENABLE_BYTERANGE)
+  set(libdispatch_SOURCES ${libdispatch_SOURCES} dhttp.c)
 ENDIF(ENABLE_BYTERANGE)
 
 IF(ENABLE_S3)
-  IF(ENABLE_S3_INTERNAL)
-    SET(libdispatch_SOURCES ${libdispatch_SOURCES} ncs3sdk_h5.c nch5s3comms.c nch5s3comms.h nccurl_sha256.c nccurl_sha256.h nccurl_hmac.c nccurl_hmac.h nccurl_setup.h)
-  ELSE()
-    SET(libdispatch_SOURCES ${libdispatch_SOURCES} ncs3sdk_aws.cpp awsincludes.h)
-  ENDIF()
-ENDIF()
+  if(ENABLE_S3_INTERNAL)
+    set(libdispatch_SOURCES ${libdispatch_SOURCES} ncs3sdk_h5.c nch5s3comms.c nch5s3comms.h nccurl_sha256.c nccurl_sha256.h nccurl_hmac.c nccurl_hmac.h nccurl_setup.h)
+  else()
+    set(libdispatch_SOURCES ${libdispatch_SOURCES} ncs3sdk_aws.cpp awsincludes.h)
+  endif()
+endif()
 
-IF(REGEDIT)
-  SET(libdispatch_SOURCES ${libdispatch_SOURCES} dreg.c)
-ENDIF(REGEDIT)
+if(REGEDIT)
+  set(libdispatch_SOURCES ${libdispatch_SOURCES} dreg.c)
+endif(REGEDIT)
 
 add_library(dispatch OBJECT ${libdispatch_SOURCES})
-IF(STATUS_PARALLEL)
+
+if (USE_HDF5)
+  target_include_directories(dispatch PRIVATE ${HDF5_INCLUDE_DIR})
+endif(USE_HDF5)
+
+if(STATUS_PARALLEL)
     target_link_libraries(dispatch PUBLIC MPI::MPI_C)
```

Comment:
This does need to be public because `MPI_Comm` appears in the signature for `nc_create_par`

Reply:
Cool, thanks for checking. Probably these would need to be scoped a bit better, since looking at `libdispatch` as a whole, the mpi/non-mpi code are intermingled.